Automattic/wp-calypso
# Gutenboarding: Fix ecommerce flow when logging in

Fix https://github.com/Automattic/wp-calypso/issues/43804 and add a few tweaks.

#### Cleanup and readability improvements
* Cleanup hooks after domain suggestions refactoring
  * Remove `getFreeDomainSuggestions` utility function.
  * Remove `useDomainSuggestions` hook.
  * Remove `freeDomainSuggestion` arg from `createSite` action.
* Define login and signup hooks correctly.
  * Rename `useOnLogin` hook to `useOnSignup` to reflect correctly its usage.
  * Extract site creation logic when logging in into a new `useOnLogin` hook.

#### Testing instructions
* Go to [/new](https://calypso.live/new?branch=update/gutenboarding-hooks-cleanup).
* Domain selection functionality should be unchanged.
* Test site creation as a new user (Signup flow) and as an existing logged-out user (Login flow).
* Test the flow choosing eCommerce plan as en existing logged-out user to check if https://github.com/Automattic/wp-calypso/issues/43804 is fixed.
